src/ImageIO/CGImageAnimation.cs

```diff
@@ -0,0 +1,131 @@
+            } finally {
+                block_block.CleanupBlock ();
+            }
+
+            return ret;
+#endif
+        }
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [BindingImpl (BindingImplOptions.Optimizable)]
+        public int AnimateImage (NSData data, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationBlock block) {
+#if IOS && ARCH_32
+            throw new PlatformNotSupportedException ();
+#else
+            if (data == null)
+                throw new ArgumentNullException (nameof (data));
+            if (block == null)
+                throw new ArgumentNullException (nameof (block));
+
+            int ret;
+            BlockLiteral block_block;
+            block_block = new BlockLiteral ();
+            block_block.SetupBlockUnsafe (SDCGImageSourceAnimationBlock.Handler, block);
+
+            try {
+                ret = CGAnimateImageDataWithBlock (data.Handle, options.GetHandle (), ref block_block);
+            } finally {
+                block_block.CleanupBlock ();
+            }
+ 
+            return ret;
+#endif
+        }
+
+        //
+        // This class bridges native block invocations that call into C#
+        //
+        static internal class SDCGImageSourceAnimationBlock {
+            static internal readonly DCGImageSourceAnimationBlock Handler = Invoke;
+
+            [MonoPInvokeCallback (typeof (DCGImageSourceAnimationBlock))]
+            static void Invoke (IntPtr block, nint index, IntPtr image, [System.Runtime.InteropServices.MarshalAs (System.Runtime.InteropServices.UnmanagedType.I1)] out bool stop) {
+                var del = BlockLiteral.GetTarget<CGImageSourceAnimationBlock>(block);
+                if (del != null)
+                    del (index, new CoreGraphics.CGImage (image), out stop);
+                else
+                    stop = default (bool);
+            }
+        } /* class SDCGImageSourceAnimationBlock */
+
+        internal sealed class NIDCGImageSourceAnimationBlock : TrampolineBlockBase {
+            DCGImageSourceAnimationBlock invoker;
+
+            [BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+            public unsafe NIDCGImageSourceAnimationBlock (BlockLiteral * block) : base (block) {
+                invoker = block -> GetDelegateForBlock<DCGImageSourceAnimationBlock> ();
+            }
+
+            [Preserve (Conditional = true)]
+            [BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+            public unsafe static CGImageSourceAnimationBlock Create (IntPtr block) {
+                if (block == IntPtr.Zero)
+                    return null;
+                var del = (CGImageSourceAnimationBlock) GetExistingManagedDelegate (block);
```

Comment:
Cant we get use the GetTarget method with the pointer here?

Reply:
I might be missing something, but I don't think this is possible. `GetTarget<CGImageSourceAnimationBlock>` returns type `CGImageSourceAnimationBlock`. 

We cannot convert from this to a `BlockLiteral`, which is needed by the constructor. The constructor will still need to call `GetDelegateForBlock` as well as the base constructor (`TrampolineBlockBase`).